modules/tinymce/src/plugins/table/main/ts/selection/TableSelection.ts

```diff
@@ -6,28 +6,33 @@
  */
 
 import { CellOpSelection, Selections, TableSelection } from '@ephox/darwin';
-import { Arr, Optionals } from '@ephox/katamari';
+import { Arr, Fun, Optionals } from '@ephox/katamari';
 import { TableLookup } from '@ephox/snooker';
 import { Attribute, Compare, SelectorFind, SugarElement, SugarElements, SugarNode } from '@ephox/sugar';
 
 import { ephemera } from './Ephemera';
 
-const getSelectionStartCellFallback = (start: SugarElement<Node>) =>
-  TableLookup.table(start).bind((table) =>
+const getSelectionCellFallback = (element: SugarElement<Node>, takeLast: boolean) =>
+  TableLookup.table(element).bind((table) =>
     TableSelection.retrieve(table, ephemera.firstSelectedSelector)
-  ).fold(() => start, (cells) => cells[0]);
+  ).fold(
+    Fun.constant(element),
+    (cells) => takeLast ? cells[cells.length - 1] : cells[0]);
 
-const getSelectionStartFromSelector = <T extends Element>(selector: string) => (start: SugarElement<Node>, isRoot?: (el: SugarElement<Node>) => boolean) => {
-  const startCellName = SugarNode.name(start);
-  const startCell = startCellName === 'col' || startCellName === 'colgroup' ? getSelectionStartCellFallback(start) : start;
-  return SelectorFind.closest<T>(startCell, selector, isRoot);
-};
+const getSelectionFromSelector = <T extends Element>(selector: string, takeLast: boolean) =>
+  (initCell: SugarElement<Node>, isRoot?: (el: SugarElement<Node>) => boolean) => {
+    const cellName = SugarNode.name(initCell);
+    const cell = cellName === 'col' || cellName === 'colgroup' ? getSelectionCellFallback(initCell, takeLast) : initCell;
+    return SelectorFind.closest<T>(cell, selector, isRoot);
+  };
+
+const getSelectionStartCaption = getSelectionFromSelector<HTMLTableCaptionElement>('caption', false);
 
-const getSelectionStartCaption = getSelectionStartFromSelector<HTMLTableCaptionElement>('caption');
+const getSelectionStartCell = getSelectionFromSelector<HTMLTableCellElement>('th,td', false);
 
-const getSelectionStartCell = getSelectionStartFromSelector<HTMLTableCellElement>('th,td');
+const getSelectionStartCellOrCaption = getSelectionFromSelector<HTMLTableCellElement | HTMLTableCaptionElement>('th,td,caption', false);
 
-const getSelectionStartCellOrCaption = getSelectionStartFromSelector<HTMLTableCellElement | HTMLTableCaptionElement>('th,td,caption');
+const getSelectionEndCellOrCaption = getSelectionFromSelector<HTMLTableCellElement | HTMLTableCaptionElement>('th,td,caption', false);
```

Comment:
The implementation of these 2 are identical. This could be changed to just a single function and renamed to `getSelectionCellOrCaption` or similar.

The same could also be done with the other 2 functions, though honestly this whole file confuses me as it's never doing anything with the actual selection. I'm not sure if you want to go that far, but at the very least please just remove the references to `Start` in this file.

- [ ] Cleanup this logic a little

Reply:
The error was a boolean that should have been true, to make it pick the last selected cell instead of the first when using the later function. Fixed that.